tools/add-test-vms.js

```diff
@@ -44,32 +43,97 @@ log = this.log = new bunyan({
-    log.debug('Morary ready!');
-
-    var nbTestVmsToCreate = Number(process.argv[2]) ||
-        DEFAULT_NB_TEST_VMS_TO_CREATE;
-    log.debug('Number of test VMs to create:', nbTestVmsToCreate);
-    assert.number(nbTestVmsToCreate);
-
-    var concurrency = Number(process.argv[3]) || DEFAULT_CONCURRENCY;
-    log.debug('concurrency:', concurrency);
-    assert.number(concurrency);
-
-    testCommon.createTestVMs(nbTestVmsToCreate, moray, {
-        concurrency: concurrency,
-        log: log
-    }, {}, function allVmsCreated(err) {
-        if (err) {
-            log.error({err: err}, 'Error when creating test VMs');
-        } else {
-            log.info('All VMs created successfully');
-        }
+var cmdlineOptions = [
+    {
+        names: ['help', 'h'],
+        type: 'bool',
+        help: 'Print this help and exit'
+    },
+    {
+        names: ['n'],
+        type: 'positiveInteger',
+        help: 'Number of test VMs to create'
+    },
+    {
+        names: ['d'],
+        type: 'string',
+        help: 'The data used to create every test VMs'
+    },
+    {
+        names: ['c'],
+        type: 'positiveInteger',
+        help: 'The number of VMs added concurrently'
+    }
+];
+
+function printUsage(cmdLineOptionsParser) {
+    var help = cmdlineOptionsParser.help({includeEnv: true}).trimRight();
+    console.log('usage: node add-test-vms.js [OPTIONS]\n' +
+        'options:\n' + help);
+}
+
+function addTestVms(nbVms, concurrency, data) {
+    assert.number(nbVms, 'nbVms must be a number');
+    assert.ok(nbVms > 0, 'nbVms must be a positive number');
+
+    assert.number(concurrency, 'concurrency must be a number');
+    assert.ok(concurrency > 0, 'concurrency must be a positive number');
+
+    assert.optionalObject(data, 'data must be an optional object');
+    data = data || {};
+
+    moray.connect();
+    moray.once('moray-ready', function () {
+        log.debug('Moray ready!');
+
+        log.debug('Number of test VMs to create:', nbTestVmsToCreate);
+        assert.number(nbTestVmsToCreate);
+
+        log.debug('concurrency:', concurrency);
+        assert.number(concurrency);
 
-        log.debug('Closing moray connection');
-        moray.connection.close();
+        testVm.createTestVMs(nbTestVmsToCreate, moray, {
+            concurrency: concurrency,
+            log: log
+        }, data, function allVmsCreated(err) {
+            if (err) {
+                log.error({err: err}, 'Error when creating test VMs');
+            } else {
+                log.info('All VMs created successfully');
+            }
+
+            log.debug('Closing moray connection');
+            moray.connection.close();
+        });
     });
-});
+}
+
+var cmdlineOptionsParser = dashdash.createParser({options: cmdlineOptions});
+
+try {
+    var parsedCmdlineOpts = cmdlineOptionsParser.parse(process.argv);
+
+    if (parsedCmdlineOpts.help) {
+        printUsage(cmdlineOptionsParser);
+        process.exitCode = 1;
+    } else {
```

Comment:
I'd be inclined to not exit non-zero for explicit `-h,--help` usage. Nit, obviously.